### PR TITLE
Introduce E2E testing

### PR DESCRIPTION
**Pull request by FinThunderstorm**
Introduce E2E testing with Playwright to help maintainability in the future. Includes also automatic accessibility testing. Added also automatic security testing.

### CODE REVIEW

**Comment by cxcorp** [comment]
I think I [just fixed](https://github.com/TKOaly/exam-archive-new/commit/814769f01f8fcaac5f428180d47b85844170b919) the docker image build CI job, could you try if you rebase to current master?

File: tests/courseList.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  await expect(row).toBeVisible()
+  await expect(row).toHaveText('Probability theory')
+  await expect(lastModified).toBeEmpty()
+})
+
+test('add a new course works', async ({ page, request }) => {
+  await page.goto('/archive')
+
+  const addBox = page.getByText(
+    'Add a new course:Create courseLogged in: dev-user (Log out)'
```
Comment (cxcorp) [inline tests/courseList.spec.ts]: Could this be targeted with a test ID? Seems like this may change pretty easily if we change the formatting of the box.

Comment (FinThunderstorm) [reply tests/courseList.spec.ts]: Changed to this. +1

**Comment by FinThunderstorm** [comment]
> I think I [just fixed](https://github.com/TKOaly/exam-archive-new/commit/814769f01f8fcaac5f428180d47b85844170b919) the docker image build CI job, could you try if you rebase to current master?

Looks like it worked, this is okay now!

File: src/index.ts [hunk 2]
```diff
@@ -29,6 +29,21 @@ app.set('view engine', 'jsx')
 app.engine('jsx', require('express-react-views').createEngine())
 app.set('trust proxy', config.TRUST_PROXY_IPS)
 
+app.disable('x-powered-by')
+app.use(function (_req, res, next) {
+  res.setHeader(
+    'Content-Security-Policy',
+    "default-src 'self' tarpisto.cdn.tko-aly.fi; img-src 'self' tarpisto.cdn.tko-aly.fi; media-src 'self' tarpisto.cdn.tko-aly.fi; script-src 'self' tarpisto.cdn.tko-aly.fi;"
```
Comment (cxcorp) [inline src/index.ts]: Not sure if cdn domain needs to be allowed at all 🤔 I think we only use it to serve the documents right now and there's no plans to start serving JS/CSS assets from there.

File: tests/accessibility.spec.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      }.html`
+    }
+  })
+  expect(accessibilityScanResults.violations).toEqual([])
+})
+
+test('examlisting check', async ({ page, browserName, isMobile }) => {
+  await page.goto('/archive/2')
+
+  const accessibilityScanResults = await new AxeBuilder({ page })
```
Comment (cxcorp) [inline tests/accessibility.spec.ts:24]: very nice

**Review by cxcorp — CHANGES_REQUESTED** [review]
Extreeemely cool, thanks so much for the contribution! I left one minor comment, otherwise looks ready to merge!

File: src/index.ts (hunk 2 above)
Comment (FinThunderstorm) [reply src/index.ts]: Dropped CDN out.

**Pull request merged**